src/tools/cephfs/MetaTool.cc

```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
+      nsop->ino_c = sop->ino;
+      op.push_op(nsop);
+      return 1;
     }else{
-        if (op.inodes.empty()){
-            meta_op::sub_op* nsop = new meta_op::sub_op(&op);
-            nsop->sub_op_t = meta_op::OP_LIST;
-            nsop->sub_ino_t = meta_op::INO_DIR;
-            nsop->trace_level = 0;
-            nsop->ino_c = sop->ino;
-            op.push_op(nsop);
-            return 1;
-        }else{
-            return -1;
-        }
+      return -1;
     }
-    return 0;
+  }
+  return 0;
 }
 int MetaTool::_show_meta(inode_meta_t& inode_meta, const string& fn){
-    std::stringstream ds;
-    std::string format = "json";
-    InodeStore& inode_data = *inode_meta.get_meta();
-    Formatter* f = Formatter::create(format);
-    f->enable_line_break();
-    f->open_object_section("meta");
-    f->dump_unsigned("snapid_t", inode_meta.get_snapid());
-    f->dump_unsigned("itype", inode_meta.get_type());
-    f->open_object_section("store");
-    inode_data.dump(f);
-    try {
-        if (inode_data.snap_blob.length()){
-            sr_t srnode;
-            auto p = inode_data.snap_blob.cbegin();
-            decode(srnode, p);
-            f->open_object_section("snap_blob");
-            srnode.dump(f);
-            f->close_section();    
-        }
-    }catch (const buffer::error &err){
-        cerr << "corrupt decode in snap_blob" 
-             << ": " << err.what() << std::endl;
-        return -1;
+  std::stringstream ds;
+  std::string format = "json";
+  InodeStore& inode_data = *inode_meta.get_meta();
+  Formatter* f = Formatter::create(format);
+  f->enable_line_break();
+  f->open_object_section("meta");
+  f->dump_unsigned("snapid_t", inode_meta.get_snapid());
+  f->dump_unsigned("itype", inode_meta.get_type());
+  f->open_object_section("store");
+  inode_data.dump(f);
+  try {
+    if (inode_data.snap_blob.length()){
+      sr_t srnode;
+      auto p = inode_data.snap_blob.cbegin();
+      decode(srnode, p);
+      f->open_object_section("snap_blob");
+      srnode.dump(f);
+      f->close_section();
+    }
+  }catch (const buffer::error &err){
+    cerr << "corrupt decode in snap_blob"
+         << ": " << err.what() << std::endl;
+    return -1;
+  }
+
+  f->close_section();
+  f->close_section();
+  f->flush(ds);
+
+  if (fn != ""){
+    ofstream o;
+    o.open(fn);
+    if (o){
+      o << ds.str();
+      o.close();
+    }
+    else{
+      cout << "out to file (" << fn << ") failed" << std::endl;
+      cout << ds.str() << std::endl;
     }
-    
-    f->close_section();
-    f->close_section();
-    f->flush(ds);
 
-    if (fn != ""){
-        ofstream o;
-        o.open(fn);
-        if (o){
-            o << ds.str();
-            o.close();
-        }
-        else{
-            cout << "out to file (" << fn << ") failed" << std::endl;
-            cout << ds.str() << std::endl;
-        }
-            
-    }else
-        std::cout << ds.str() << std::endl;
-    return 0;
+  }else
```

Comment:
space.

Reply:
@tchaikov updated it.

Reply:
@tchaikov
https://github.com/ceph/ceph/pull/32422
https://github.com/ceph/ceph/pull/31514
These two patches haven't progressed for two months. review them please.
I'll fix them as soon as possible if there's a problem.
thx.